### PR TITLE
Feat/company show

### PR DESCRIPTION
**Pull request by MDelarosa1993**
## Type of Change
- [x] feature ⛲
- [x] bug fix 🐛
- [x] refactor 🧑‍💻
- [x] testing 🧪

## Description
This PR introduces several enhancements and fixes, including:
- Adding a show route for Company with nested Contact resources
- Updating the ContactsController to handle cases when there are no contacts associated with a user or a specific company
- Enhancing policy checks for viewing and listing companies to ensure proper authorization
- Refactoring redundant code in controllers and serializers to improve readability and maintainability
- Writing additional request specs for happy paths and edge cases in Company and Contact resources

## Motivation and Context
- To improve the functionality and security of the Company and Contact features
- Allows users to fetch Contact data based on their associated Company
- Ensures proper authorization and error handling for edge cases

## Added Test?
- [x] Yes 🫡
 Request/Controller Specs
- [x] All previous tests still pass 🥳
- User spec still have failing tests for too much rendering.

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: app/controllers/api/v1/contacts_controller.rb [hunk 1]
```diff
@@ -4,12 +4,37 @@ class ContactsController < ApplicationController
       before_action :authenticate_user
 
       def index
-				authorize Contact
```
Comment (stefanjbloom) [inline app/controllers/api/v1/contacts_controller.rb:7]: A lot of thought and work this refractor. Super precise way for this specific action. 

However lemme be the Devil's advocate: I could potentially see this logic moved into the model in the future. Then those methods called here. As of now, it bulks the controller.

File: spec/requests/api/v1/companies/show_spec.rb
Comment (Sgalvin36) [inline spec/requests/api/v1/companies/show_spec.rb:1]: Very thorough testing. Even have some sad path tests. For the associated contacts to a company, does it return any associated contacts? or just the ones tied to the specific user? May be a good edge case to test.

**Review by Sgalvin36 — APPROVED** [review]
Looks good overall. Several questions of potential refactors but nothing that should prevent it being added to main now.

File: app/policies/company_policy.rb [hunk 2]
```diff
@@ -3,8 +3,13 @@ def create?
     user.present?
   end
 
+  def show?
+    return false unless user
```
[the remark this answers is not in the document]
Comment (jimmacur) [reply app/policies/company_policy.rb:7]: The first line efficiently handles guest users by immediately returning false if no user is present. The second line then checks if the user is an admin or the owner of the record.

File: spec/requests/api/v1/companies/show_spec.rb
Comment (MDelarosa1993) [reply spec/requests/api/v1/companies/show_spec.rb:1]: Correct me if im wrong but I think we have that test in the show_spec file. Will double check for it

Comment (MDelarosa1993) [reply spec/requests/api/v1/companies/show_spec.rb:1]: Only associated contacts to the company

File: app/serializers/user_serializer.rb
[the remark this answers is not in the document]
Comment (MDelarosa1993) [reply app/serializers/user_serializer.rb:1]: Yeah, Having the companies attribute there was always showing companies for a user, even though we weren't hitting the companies endpoint.

File: spec/requests/api/v1/companies/show_spec.rb
Comment (jimmacur) [reply spec/requests/api/v1/companies/show_spec.rb:1]: For the test on line 62, we show that only contacts associated to a company and user will be returned. 

Since companies and contacts are not global there could not be a contact that is associated to the company but is not associated to the user.
